pkgs/by-name/gh/ghostty/package.nix

```
@@ -0,0 +1,139 @@
+{
+  lib,
+  stdenv,
+  bzip2,
+  callPackage,
+  expat,
+  fetchFromGitHub,
+  fontconfig,
+  freetype,
+  git,
+  glib,
+  gsettings-desktop-schemas,
+  gtk4,
+  harfbuzz,
+  libadwaita,
+  libGL,
+  libpng,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  ncurses,
+  nix-update-script,
+  oniguruma,
+  pandoc,
+  pkg-config,
+  wrapGAppsHook4,
+  zig_0_13,
+  zlib,
+}:
+let
+  # Ghostty needs to be built with --release=fast, --release=debug and
+  # --release=safe enable too many runtime safety checks.
+  zig_hook = zig_0_13.hook.overrideAttrs {
+    zig_default_flags = "-Dcpu=baseline -Doptimize=ReleaseFast --color off";
+  };
+in
+stdenv.mkDerivation (final: {
+  pname = "ghostty";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ghostty-org";
+    repo = "ghostty";
+    tag = "v${final.version}";
+    hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
+  };
+
+  nativeBuildInputs = [
+    git
+    ncurses
+    pandoc
+    pkg-config
+    wrapGAppsHook4
+    zig_hook
+  ];
+
+  buildInputs = [
+    bzip2
+    expat
+    fontconfig
+    freetype
+    glib
+    gsettings-desktop-schemas
+    gtk4
+    harfbuzz
+    libadwaita
+    libGL
+    libpng
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    oniguruma
+    zlib
+  ];
+
+  dontPatch = true;
+  dontConfigure = true;
+  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
+  doCheck = false;
+
+  deps = callPackage ./deps.nix { name = "${final.pname}-cache-${final.version}"; };
+
+  zigBuildFlags = [
+    "--system"
+    "${final.deps}"
+    "-Dversion-string=${final.version}"
+  ];
+
+  zigCheckFlags = final.zigBuildFlags;
+
+  outputs = [
+    "out"
+    "terminfo"
+    "shell_integration"
+    "vim"
+  ];
+
+  postInstall = ''
+    mkdir -p "$terminfo/share"
+    mv "$out/share/terminfo" "$terminfo/share/terminfo"
+    ln -sf "$terminfo/share/terminfo" "$out/share/terminfo"
+
+    mkdir -p "$shell_integration"
+    mv "$out/share/ghostty/shell-integration" "$shell_integration/shell-integration"
+    ln -sf "$shell_integration/shell-integration" "$out/share/ghostty/shell-integration"
+
+    mv "$out/share/vim/vimfiles" "$vim"
+    ln -sf "$vim" "$out/share/vim/vimfiles"
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(--prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libX11 ]}")
+  '';
```

```suggestion
  NIX_LDFLAGS = [ "-lX11" ];
```
We can avoid this hack by forcing linkage at compile time

```console
$ ldd result/bin/ghostty
        linux-vdso.so.1 (0x00007ffd11ade000)
        libX11.so.6 => /nix/store/nlqind4szw3amcmhgy4pd2n0894558gg-libX11-1.8.10/lib/libX11.so.6 (0x00007fce9daea000)
        libc.so.6 => /nix/store/65h17wjrrlsj2rj540igylrx7fqcd6vq-glibc-2.40-36/lib/libc.so.6 (0x00007fce9d8e5000)
        libpthread.so.0 => /nix/store/65h17wjrrlsj2rj540igylrx7fqcd6vq-glibc-2.40-36/lib/libpthread.so.0 (0x00007fce9d8e0000)
        libxcb.so.1 => /nix/store/2j3c18398phz5c1376x2qvva8gx9g551-libxcb-1.17.0/lib/libxcb.so.1 (0x00007fce9d8b3000)
        /nix/store/65h17wjrrlsj2rj540igylrx7fqcd6vq-glibc-2.40-36/lib/ld-linux-x86-64.so.2 => /nix/store/wn7v2vhyyyi6clcyn0s9ixvl7d4d87ic-glibc-2.40-36/lib64/ld-linux-x86-64.so.2 (0x00007fce9dc3c000)
        libXau.so.6 => /nix/store/hjbxiwsc587b8dc6v6pisa34aj10hq23-libXau-1.0.11/lib/libXau.so.6 (0x00007fce9d8ac000)
        libXdmcp.so.6 => /nix/store/c9gk656q2x8av467r06hcjag31drjfzh-libXdmcp-1.1.5/lib/libXdmcp.so.6 (0x00007fce9d8a4000)
```